machines/biometrics.ts

```diff
@@ -0,0 +1,227 @@
+import { createModel } from "xstate/lib/model";
+import * as LocalAuthentication from 'expo-local-authentication';
+import { assign, send, StateFrom } from "xstate";
+
+
+// --- CREATE MODEL -----------------------------------------------------------
+const model = createModel(
+  {
+    isAvailable : false,
+    authTypes   : [],
+    isEnrolled  : false,
+    status      : null,
+    retry       : false,
+  },
+  {
+    events: {
+      AUTHENTICATE: () => ({}),
+      RETRY_AUTHENTICATE: () => ({})
+    }
+  }
+);
+// ----------------------------------------------------------------------------
+
+// --- CREATE MACHINE ---------------------------------------------------------
+export const biometricsMachine = model.createMachine(
+  {
+    id: 'biometrics',
+    initial: 'init',
+    states: {
+
+      // Initializing biometrics states
+      init: {
+        invoke: {
+          src: LocalAuthentication.hasHardwareAsync,
+          onError: 'failure',
+          onDone: {
+            target: 'initAuthTypes',
+            actions: assign({ isAvailable: (context, event) => event.data }),
```

Comment:
Please use named actions so it's easier to read the state machine. Ideally the state definition is JSON-serializable so we can also take advantage of using the visualizers. Another benefit is allowing us to reduce code duplication if we reuse an action (or even an entire state since we can easily compose objects).

Reply:
Also use `model.assign` for typings.

Reply:
Ohhhhh i see, ill do that thanks